### PR TITLE
Fix GCSToBigQueryOperator not respecting schema_obj

### LINKED ISSUE #28441: GCSToBigQueryOperator fails when schema_object is specified without schema_fields
### Apache Airflow Provider(s)

google

### Versions of Apache Airflow Providers

apache-airflow 2.5.0
apache-airflow-providers-apache-beam 4.1.0
apache-airflow-providers-cncf-kubernetes 5.0.0
apache-airflow-providers-google 8.6.0
apache-airflow-providers-grpc 3.1.0

### Apache Airflow version

2.5.0

### Operating System

Debian 11

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

KubernetesExecutor

### What happened

GCSToBigQueryOperator allows multiple ways to specify schema of the BigQuery table:

1. Setting autodetect == True
1. Setting schema_fields directly with autodetect == False
1. Setting a schema_object and optionally a schema_object_bucket with autodetect == False

This third method seems to be broken in the latest provider version (8.6.0) and will always result in this error:

```
[2022-12-16, 21:06:18 UTC] {taskinstance.py:1772} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/google/cloud/transfers/gcs_to_bigquery.py", line 395, in execute
    self.configuration = self._check_schema_fields(self.configuration)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/google/cloud/transfers/gcs_to_bigquery.py", line 524, in _check_schema_fields
    raise RuntimeError(
RuntimeError: Table schema was not found. Set autodetect=True to automatically set schema fields from source objects or pass schema_fields explicitly
```

The reason for this is because [this block](https://github.com/apache/airflow/blob/25bdbc8e6768712bad6043618242eec9c6632618/airflow/providers/google/cloud/transfers/gcs_to_bigquery.py#L318-L320) where `if self.schema_object and self.source_format != "DATASTORE_BACKUP":`. fails to set self.schema_fields. It only sets the local variable, schema_fields. When self._check_schema_fields is subsequently called [here](https://github.com/apache/airflow/blob/25bdbc8e6768712bad6043618242eec9c6632618/airflow/providers/google/cloud/transfers/gcs_to_bigquery.py#L395), we enter the [first block](https://github.com/apache/airflow/blob/25bdbc8e6768712bad6043618242eec9c6632618/airflow/providers/google/cloud/transfers/gcs_to_bigquery.py#L523-L528) because autodetect is false and schema_fields is not set.

### What you think should happen instead

No error should be raised if autodetect is set to False and a valid schema_object is provided

### How to reproduce

1. Create a simple BigQuery table with a single column col1:

```sql
CREATE TABLE `my-project.my_dataset.test_gcs_to_bigquery` (col1 INT);
```

2. Upload a json blob for this object to a bucket (e.g., data/schemas/table.json)
3. Upload a simple CSV for the source file to load to a bucket (e.g., data/source/file.csv)
4. Run the following command:

```py
    gcs_to_biquery = GCSToBigQueryOperator(
        task_id="gcs_to_bigquery",
        destination_project_dataset_table="my-project.my_dataset.test_gcs_to_bigquery",
        bucket="my_bucket_name",
        create_disposition="CREATE_IF_NEEDED",
        write_disposition="WRITE_TRUNCATE",
        source_objects=["data/source/file.csv"],
        source_format="CSV",
        autodetect=False,
        schema_object="data/schemas/table.json",
    )
```

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by vchiapaikeo**
GCSToBigQueryOperator allows multiple ways to specify schema of the BigQuery table:

1. Setting autodetect == True
1. Setting schema_fields directly with autodetect == False
1. Setting a schema_object and optionally a schema_object_bucket with autodetect == False

This third method seems to be broken in the latest provider version (8.6.0) and will always result in this error:

```
[2022-12-16, 21:06:18 UTC] {taskinstance.py:1772} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/google/cloud/transfers/gcs_to_bigquery.py", line 395, in execute
    self.configuration = self._check_schema_fields(self.configuration)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/google/cloud/transfers/gcs_to_bigquery.py", line 524, in _check_schema_fields
    raise RuntimeError(
RuntimeError: Table schema was not found. Set autodetect=True to automatically set schema fields from source objects or pass schema_fields explicitly
```

The reason for this is because [this block](https://github.com/apache/airflow/blob/25bdbc8e6768712bad6043618242eec9c6632618/airflow/providers/google/cloud/transfers/gcs_to_bigquery.py#L318-L320) where `if self.schema_object and self.source_format != "DATASTORE_BACKUP":` fails to set self.schema_fields. It only sets the local variable, schema_fields. When self._check_schema_fields is subsequently called [here](https://github.com/apache/airflow/blob/25bdbc8e6768712bad6043618242eec9c6632618/airflow/providers/google/cloud/transfers/gcs_to_bigquery.py#L395), we enter the [first block](https://github.com/apache/airflow/blob/25bdbc8e6768712bad6043618242eec9c6632618/airflow/providers/google/cloud/transfers/gcs_to_bigquery.py#L523-L528) because autodetect is false and self.schema_fields is not set.

This PR sets the instance variable, self.schema_fields when the user passes in a schema_obj. Additionally, it uses self.schema_object_bucket instead of the erroneous self.bucket.

cc: @eladkal
Fixes: https://github.com/apache/airflow/issues/28441

### CODE REVIEW

File: airflow/providers/google/cloud/transfers/gcs_to_bigquery.py [hunk 1]
```diff
@@ -320,8 +320,10 @@ def execute(self, context: Context):
                 impersonation_chain=self.impersonation_chain,
             )
             if self.schema_object and self.source_format != "DATASTORE_BACKUP":
-                schema_fields = json.loads(gcs_hook.download(self.bucket, self.schema_object).decode("utf-8"))
-                self.log.info("Autodetected fields from schema object: %s", schema_fields)
+                self.schema_fields = json.loads(
+                    gcs_hook.download(self.schema_object_bucket, self.schema_object).decode("utf-8")
+                )
+                self.log.info("Autodetected fields from schema object: %s", self.schema_fields)
```
Comment (eladkal) [inline airflow/providers/google/cloud/transfers/gcs_to_bigquery.py]: Correct me if I'm wrong but the entire block lines of 316-326 can be replaced with same code as we have in 
https://github.com/apache/airflow/blob/199359bb3886699904ca075de7bd5fdfe5105c5f/airflow/providers/google/cloud/operators/bigquery.py#L1603-L1611

which is shorter and with no nesting if

Comment (vchiapaikeo) [reply airflow/providers/google/cloud/transfers/gcs_to_bigquery.py]: Yes, good point.

**Comment by vchiapaikeo** [comment]
Whoops @eladkal - fixed the isort failure

**Review by eladkal — APPROVED** [review]

**Comment by vchiapaikeo** [comment]
cc @turbaszek, I believe you are codeowner. Can you approve this as well?

**Comment by eladkal** [comment]
Thanks @vchiapaikeo !
If you have time I'd appreciate also helping to resolve https://github.com/apache/airflow/issues/12329

**Comment by vchiapaikeo** [comment]
My pleasure @eladkal! Sure I can try to take a look at it this weekend.

**Comment by axelborja** [comment]
Any idea in which release this one will be shipped ?

**Comment by potiuk** [comment]
Screenshot 2023-01-17 at 12 35 50

Seen the above ? ^^ looks like this is already shipped in the previous provider and you can update the provider when you want.

**Comment by grimwm** [comment]
Hi, when will this be released? It is not in 2.5.1.

**Comment by potiuk** [comment]
> Hi, when will this be released? It is not in apache-airflow[gcp]==2.5.1.

Why do you think so ?

**Comment by grimwm** [comment]
> > Hi, when will this be released? It is not in apache-airflow[gcp]==2.5.1.
> 
> Why do you think so ?

Hi, that was my fault. I didn't realize our production Airflow was on 2.5.0. I finally put it together and got it sorted and forgot to update here. Sorry, and thanks!

**Pull request merged**